config/config.go

```diff
@@ -1083,7 +1083,46 @@ type Config struct {
 		Vault  VaultConfig  `json:"vault"`
 	} `json:"kv"`
 
-	// Secrets are key-value pairs that can be accessed in the dashboard via "secrets://"
+	// The TYK_GW_SECRETS environment variable allows you to configure the "secrets"
+	// section in the Tyk Gateway configuration file. These secrets are key-value pairs
+	// that can be dynamically injected into API Definitions and the tyk.conf configuration
+	// file using the secrets://KEY notation. For Gateway versions prior to 5.3.0 only the
+	// listen path and target URL fields can use this notation.
+	//
+	// Syntax: TYK_GW_SECRETS=key1:/value1,key2:/value2
+	// Example: TYK_GW_SECRETS=api-listen-path:/secret,api-auth-key:/another-secret
```

Comment:
@lghiur @rewsmith Is the example provided here correct? Do we need `/` after the colon `:` for specifying the value of the secret? Because the [docs](https://tyk.io/docs/tyk-configuration-reference/kv-store/#environment-variables) don't specify it. 

And it seems misleading.

Reply:
The example is technically correct if you consider `/secret` to be the value. The kingpin/v2 library is short on what it does in this case:

> StringMap provides key=value parsing into a map.

Imo except for listen path the `/` should be removed, or a more demonstrative example given here.

Reply:
Perhaps give the syntax as:
`TYK_GW_SECRETS=<key1>:<value1>, <key2>:<value2>`

...and then a real example that *could* include a leading `/` if that's useful

Reply:
Keep in mind, as cited above the delimiter is `=` not `:` so that should also be adjusted from `:`

Reply:
@titpetric So is this correct? `TYK_GW_SECRETS=<key1>=<value1>,<key2>=<value2>`

Reply:
Sorry it seems I was wrong on that one, the `:` as the delimiter value is the correct case. It's now covered with a unit test that also demonstrates that `/` is part of the value. See `TestSecretsDecoder`.